### PR TITLE
Generator trait produces incorrect exception message for invalid directory paths

### PR DESCRIPTION
**Pull request by sjokkateer**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description
Within the `GeneratorTrait::ensureDirectory` it is possible to get back an exception message containing an incorrect directory name. This test plus modification should ensure the directory that could not be created is returned in the error message.

### CODE REVIEW

File: test/CodeGenerator/GeneratorTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use PHPUnit\Framework\TestCase;
+
+final class GeneratorTraitTest extends TestCase
+{
+    public function testEnsureDirectoryGivenInvalidDirectoryNameExpectedErrorMessageContainInvalidDirectoryName(): void
+    {
+        $invalidDir = 'http://www.invalid-directory';
+
+        $this->expectException(GenerateCodeException::class);
+        $this->expectErrorMessage("Could not create output directory: {$invalidDir}");
```
Comment (Ocramius) [inline test/CodeGenerator/GeneratorTraitTest.php]: Please hardcode or concatenate the value here: string interpolation is a mess to read 😁

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/CodeGenerator/GeneratorTraitTest.php (hunk 1 above)
Comment (sjokkateer) [reply test/CodeGenerator/GeneratorTraitTest.php]: @Ocramius Will do, I'll also fix my commits in a different request.

Comment (Ocramius) [reply test/CodeGenerator/GeneratorTraitTest.php]: You just need to send the fix-up commits to the same branch

**Review by Ocramius — APPROVED** [review]
Thanks @sjokkateer!

**Pull request merged**